From a3d55c1f983d2470d1d0587445d91fc8a5a1052c Mon Sep 17 00:00:00 2001
From: Nikolay Nechaev <nikolay_nechaev@mail.ru>
Date: Sat, 29 Jun 2024 18:43:58 +0300
Subject: [PATCH] Use the absolute real path to a video for watch_later

---
 player/configfiles.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/player/configfiles.c b/player/configfiles.c
index 418937b11d..ad24137c6e 100644
--- a/player/configfiles.c
+++ b/player/configfiles.c
@@ -16,6 +16,7 @@
  */
 
 #include <errno.h>
+#include <stdlib.h>
 #include <stddef.h>
 #include <stdbool.h>
 #include <sys/types.h>
@@ -220,7 +221,7 @@ static char *mp_get_playback_resume_config_filename(struct MPContext *mpctx,
     } else if (opts->ignore_path_in_watch_later_config) {
         path = mp_basename(fname);
     } else {
-        path = mp_normalize_path(tmp, fname);
+        path = realpath(fname, NULL);
         if (!path)
             goto exit;
     }
-- 
2.49.0

